core-bundle/src/EventListener/DataContainer/AdjustGalleryPaletteListener.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\ContentGallery;
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+
+/**
+ * Adjusts the palette for the "gallery" content element when the legacy version
+ * is in use.
+ */
+#[AsCallback('tl_content', 'config.onload')]
+class AdjustGalleryPaletteListener
```

Comment:
Should we maybe call it `LegacyGalleryPaletteListener`? Then it's already obvious by the filename, that this is only for the old stuff. 🙃

Reply:
Shouldn't we also add a deprecation notice to not forget to remove it in Contao 6 😃?

Reply:
The listener is not part of the public API. We can add a "todo", though. But @leofeyer will probably handle this when merging, right? ☺️